lib/spack/spack/solver/asp.py

```diff
@@ -98,9 +99,38 @@ def getter(node):
     'DeclaredVersion', ['version', 'idx', 'origin']
 )
 
+
+# Class for negating version ranges
+# This allows us to sort ascending by inverse version order
+class _InvertedVersion(object):
+    def __init__(self, version):
+        self.version = version
+
+    def __eq__(self, other):
+        return self.version == other.version
+
+    def __lt__(self, other):
+        return self.version > other.version
+
+    def __gt__(self, other):
+        return self.version < other.version
+
+    def __ne__(self, other):
+        return self.version != other.version
+
+    def __le__(self, other):
+        return self.version >= other.version
+
+    def __ge__(self, other):
+        return self.version <= other.version
```

Comment:
Can we use this instead:
```suggestion
@functools.total_ordering
class _InvertedVersion(object):
    def __init__(self, version):
        self.version = version

    def __eq__(self, other):
        return self.version == other.version

    def __lt__(self, other):
        return self.version > other.version
```
? Or are there performance issues with the decorator?